drivers/gpio/gpio_stm32.c

```diff
@@ -288,8 +337,11 @@ static int gpio_stm32_config(struct device *dev, int access_op,
 		return -EIO;
 	}
 
-	if (IS_ENABLED(CONFIG_EXTI_STM32) && (flags & GPIO_INT) != 0) {
+	if (!IS_ENABLED(CONFIG_EXTI_STM32)) {
+		goto release_lock;
+	}
 
+	if (flags & GPIO_INT) {
 		if (stm32_exti_set_callback(pin, cfg->port,
 					    gpio_stm32_isr, dev) != 0) {
 			return -EBUSY;
```

Comment:
It doesn't quite belong to this PR, but also here we should have `goto release_lock;`. Maybe we should have an extra commit that cleans up this part of the code and ensures that hardware semaphore on dual core SoCs is always correctly released?

Reply:
@mnkp, indeed. I took care of this gpio driver update to new API, but would be cleaner if done here.

Reply:
I've pushed a dedicated PR for this purpose, you can rebase on top of it: https://github.com/zephyrproject-rtos/zephyr/pull/19646